libyaml_q_declaration.md

```diff
@@ -0,0 +1,124 @@
+# External Dependency Quality declaration libyaml
+
+This document is a declaration of software quality for the `libyaml` external dependency, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/rep-2004/rep-2004.rst).
+
+The [libyaml](https://github.com/yaml/libyaml) external dependency is a C library for parsing and emitting YAML. Maintained in the YAML Project Github organization together with other 115 repositories. First, a summary discussing how this library is qualified is presented, and then it will be listed how this library matches the standards defined for ROS packages.
+
+## Summary
+
+The `libyaml` meets the basic requirements for a software platform in terms of testing its basic functionality, providing a valid license for the code used and a public github repository with the changes made to the code over time.
+
+Even if the library does not provide an API/ABI policy targeting the desired use of the library, the fact that it deals with the YAML standard and this one hasn’t changed since 2009, allows us to infer that the functionality needed for the ROS core from this library is not going to be changed.
+
+There is no explicit support for any OS platform, however their [Github repository](https://github.com/yaml/libyaml) installation appears to be targeting Linux. The first version of this library was developed in 2006, and it is used widely. There is no explicit metric of how much the library is used, but the equivalent library for python, developed by the same organization is required for at least 150k repositories (According to [Github metrics](https://github.com/yaml/pyyaml/network/dependents?package_id=UGFja2FnZS01MjUyMjEzNQ%3D%3D)) and the `libyaml` library is used for some optional fast functionality. The [safe_yaml](https://rubygems.org/gems/safe_yaml) ruby gem has over 80million downloads and one of its implementations uses `libyaml` through psych. There are various big projects leveraging functionality from the `libyaml` library.
+
+To study the library as an external dependency, it might not be necessary to require high code coverage, but to check if the used parts in the ROS packages are tested, and this is the case for this library.
```

Comment:
can we give some pointer? (maybe to the system_test repo)

Reply:
Yes, I'll check for them early next week. Although when I wrote initially this section I was thinking in some sort of unit testings included in the `libyaml_vendor` package. Although I'm not sure if this is feasible or if we will aim to do that.

Reply:
@ahcorde Been checking the tests and seems this is actually not the case, I'll remove this paragraph.